presto-main/src/main/java/io/prestosql/memory/QueryContext.java

```diff
@@ -110,13 +114,26 @@ public QueryContext(
                 newRootAggregatedMemoryContext(new QueryMemoryReservationHandler(this::updateSystemMemory, this::tryReserveMemoryNotSupported), 0L));
     }
 
+    public boolean isMemoryLimitsInitialized()
+    {
+        return memoryLimitsInitialized;
+    }
+
+    public void setMemoryLimitsInitialized()
+    {
+        memoryLimitsInitialized = true;
+    }
+
     // TODO: This method should be removed, and the correct limit set in the constructor. However, due to the way QueryContext is constructed the memory limit is not known in advance
     public synchronized void setResourceOvercommit()
     {
+        resourceOverCommit = true;
         // Allow the query to use the entire pool. This way the worker will kill the query, if it uses the entire local general pool.
         // The coordinator will kill the query if the cluster runs out of memory.
         maxUserMemory = memoryPool.getMaxBytes();
         maxTotalMemory = memoryPool.getMaxBytes();
+        //  Mark future memory limit updates as unnecessary
+        setMemoryLimitsInitialized();
```

Comment:
this is redundant. We already call `queryContext.setMemoryLimitsInitialized();` in `updateTask`

Reply:
Sure.